hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -2061,12 +2060,7 @@ public DatanodeInfo[] load(DatanodeReportType type) throws Exception {
     public ListenableFuture<DatanodeInfo[]> reload(
         final DatanodeReportType type, DatanodeInfo[] oldValue)
         throws Exception {
-      return executorService.submit(new Callable<DatanodeInfo[]>() {
-        @Override
-        public DatanodeInfo[] call() throws Exception {
-          return load(type);
-        }
-      });
+      return executorService.submit(() -> load(type));
```

Comment:
One of the reasons to not use this syntax was to keep some backwards compatibility with branch-2.
I guess that's long gone so let's go ahead.